### DISCUSSION TITLE
Characteristic without write-without-response property

### DISCUSSION THREAD

**Discussion by jedras1991** [Jakeler/ble-serial#27] [Q&A] [answered]
Hello,
my BT module which is [Bluno nano](https://github.com/keywish/keywish-nano-plus/blob/master/BLE-Nano/Ble-Nano%20operation%20manual%20V.1.6.pdf) does not have write-without-response property , instead it has write property.
![image](https://user-images.githubusercontent.com/14330747/118811798-b5918d00-b8ad-11eb-8660-247e77405b9e.png)

Connecting gives me this error:
![image](https://user-images.githubusercontent.com/14330747/118820882-f7730100-b8b6-11eb-9368-b9b8953726cf.png)

Is it still possible to connect with this device?

**Comment by jedras1991** [comment]
Changed ` self.write_char = self.find_char(write_uuid, 'write-without-response')` to ` self.write_char = self.find_char(write_uuid, 'write')` in `ble_interface.py` and it worked!

**Reply by Jakeler** [reply]
Very nice, thanks for looking into it and trying it out. 
I will soon improve the characteristic detection/handling (#28) and also take the `write` property into account.

**Comment by Jakeler** [comment, marked as answer]
I have just released [v2.1.0](https://github.com/Jakeler/ble-serial/releases/tag/v2.1.0) that works with both `write` and `write-without-response`!
